Add config flow for Derivative (#22063)
home-assistant/home-assistant.github.io: source/_integrations/derivative.markdown

ha_codeowners:
  - '@afaucogney'
ha_domain: derivative
ha_config_flow: true
ha_platforms:
  - sensor
---

The derivative integration creates a sensor that estimates the derivative of the

> values provided by another source sensor. Derivative sensors are updated upon changes of the **source sensor** used.

Reviewer: ```suggestion
values provided by another sensor (the **source sensor**). Derivative sensors are updated upon changes of the **source sensor**.
```

{% include integrations/config_flow.md %}
{% configuration_basic %}
Name:
  description: The name the sensor should have. You can change it again later.

> Input sensor:

Reviewer: Should we call it the "source sensor" instead?

Author: 🤷  I think of it as an input (which matched terminology we use with Blueprints I guess). But, either way, is fine IMHO. Currently, it matches the Core PR.

  description: The entity providing numeric readings to create the derivative of.
Precision:
  description: Round the calculated integration value to at most N decimal places.
Time window:

>   description: The time window in which to calculate the derivative. Derivatives in this window will be averaged with a Simple Moving Average algorithm weighted by time. This is for instance useful for a sensor that outputs discrete values, or to filter out short duration noise. By default the derivative is calculated between two consecutive updates without any smoothing.

Reviewer: ```suggestion
  description: The time window in which to calculate the derivative. Derivatives in this window will be averaged with a simple moving average algorithm (SMA) weighted by time. This is for instance useful for a sensor that outputs discrete values, or to filter out short duration noise. By default the derivative is calculated between two consecutive updates without any smoothing.
```

Metric Prefix:
  description: Metric unit to prefix the derivative result ([Wikipedia](https://en.wikipedia.org/wiki/Unit_prefix)).

> Time unoit:

Reviewer: ```suggestion
Time unit:
```

>   description: SI unit of time of the derivative. If this parameter is set, the unit of measurement will be set like x/y where x is the unit of the input sensor given and y is the value given here.

Reviewer: ```suggestion
  description: SI unit of time of the derivative. If this parameter is set, the unit of measurement will be set to **x/y** where **x** is the unit of the source sensor and **y** is the value of this parameter.
```
